sdk/test-utils/perf/src/utils.ts

```diff
@@ -100,3 +100,29 @@ export function formatDuration(durationMilliseconds: number): string {
 
   return `${minutes < 10 ? "0" : ""}${minutes}:${seconds < 10 ? "0" : ""}${seconds}`;
 }
+
+// Formats a number with a minimum number of significant digits.
+// Digits to the left of the decimal point are always significant.
+// Examples:
+// - formatNumber(0, 4) -> "0.000"
+// - formatNumber(12345, 4) -> "12,345"
+// - formatNumber(1.2345, 4) -> "1.235"
+// - formatNumber(0.00012345, 4) -> "0.0001235"
+export function formatNumber(value: number, minSignificantDigits: number) {
```

Comment:
I wouldn't say we are "hand-rolling our own function" here.  `value.toLocaleString()` is doing the heavy lifting.  We need our own function because we want digits to the left of the decimal point to always be significant.  This goes against the strict definition of "significant digits", but we thought it would be confusing to drop digits left of the decimal point (e.g all 5-digit values would end in zero).

I don't feel strongly about delegating to `Number.toLocaleString()` vs `Number.toPrecision()`, but I think `toLocaleString()` is better for this purpose.

Reply:
In .NET, Java, and JS we did need to hand-roll the rounding algorithm, since these languages don't have significant digit formatting built-in.  Example:

https://github.com/Azure/azure-sdk-for-net/pull/30170/files#diff-5325d55a387a798138d92e9a1bbdea70ee99aae7ebe90ceef4143e26c1c97ba1R30

Reply:
Ah, OK, gotcha, that makes sense. I was hoping we could drop `formatNumber` entirely and replace it with `Number.toPrecision`, but I guess we can't.  It looks like `toPrecision` falls back to scientific notation (e.g. `(12345).toPrecision(4)` gives `1.23e+4`), which is no good either.

All looks good to me then.